salt/states/win_wua.py

```diff
@@ -486,25 +486,36 @@ def uptodate(
 
     post_info = wua.updates().list()
 
+    # First gather list of updates that weren't installed
+    updates_not_installed = [item for item in install.list() if item not in post_info]
+
     # Verify the installation
     for item in install.list():
-        if not salt.utils.data.is_true(post_info[item]["Installed"]):
-            ret["changes"]["failed"] = {
-                item: {
-                    "Title": post_info[item]["Title"][:40] + "...",
-                    "KBs": post_info[item]["KBs"],
+        # Only check for items that were installed
+        if item not in updates_not_installed:
+            if not salt.utils.data.is_true(post_info[item]["Installed"]):
+                ret["changes"]["failed"] = {
+                    item: {
+                        "Title": post_info[item]["Title"][:40] + "...",
+                        "KBs": post_info[item]["KBs"],
+                    }
                 }
-            }
-            ret["result"] = False
-        else:
-            ret["changes"]["installed"] = {
-                item: {
-                    "Title": post_info[item]["Title"][:40] + "...",
-                    "NeedsReboot": post_info[item]["NeedsReboot"],
-                    "KBs": post_info[item]["KBs"],
+                ret["result"] = False
+            else:
+                ret["changes"]["installed"] = {
+                    item: {
+                        "Title": post_info[item]["Title"][:40] + "...",
+                        "NeedsReboot": post_info[item]["NeedsReboot"],
+                        "KBs": post_info[item]["KBs"],
+                    }
                 }
-            }
 
+    # Add the list of updates not installed to the return
+    if len(updates_not_installed) > 0:
```

Comment:
I wonder lint said you nothing about his. Please change to `if updates_not_installed:`